src/python/pants/engine/target.py

```diff
@@ -642,6 +649,23 @@ def validate(self) -> None:
         """
 
 
+def _validate_origin_text_blocks(origin_text_blocks: FrozenDict[str, TextBlocks]) -> None:
+    if not isinstance(origin_text_blocks, FrozenDict):
+        raise ValueError(
+            f"Expected type `FrozenDict`, got {type(origin_text_blocks)=} {origin_text_blocks=}"
+        )
+    for blocks in origin_text_blocks.values():
+        if not isinstance(blocks, tuple):
+            raise ValueError(
+                f"Expected `FrozenDict` values to be of type `tuple[TextBlock, ...]`, got {type(blocks)=} {blocks=}"
```

Comment:
I think this error message would be confusing to get as it doesn't mention what argument this value relates to...

```suggestion
                f"Expected `origin_text_blocks` to be a `FrozenDict` with values of type `TextBlocks`, got values of {type(blocks)=} {blocks=}"
```